### DISCUSSION TITLE
upgraded all npm packages (inc adonis auth, shield,..)

### DISCUSSION THREAD

**Discussion by WeyIin** [adonisjs/core#1711] [Help] [answered]
I just upgraded all my NPM and now I am facing this issue:

```
×  error       typescript compiler errors
app/Middleware/Auth.ts:27:26 - error TS2345: Argument of type 'string' is not assignable to parameter of type '"
web"'.

27       if (await auth.use(guard).check()) {
                            ~~~~~
config/shield.ts:59:3 - error TS2322: Type '{ enabled: false; directives: {}; loose: boolean; reportOnly: false;
 setAllHeaders: boolean; disableAndroid: boolean; }' is not assignable to type 'CspOptions'.
  Object literal may only specify known properties, and 'loose' does not exist in type 'CspOptions'.

59   loose: false,
```

It is preventing user to log in or register a new user.

edit:

When registering or seeding an user:

```
❯ error     database\seeders\Account
  Cannot find module 'phc-bcrypt'
Require stack:
- node_modules\@adonisjs\hash\build\src\Drivers\Bcrypt.js
- node_modules\@adonisjs\hash\build\src\Hash\index.js
- node_modules\@adonisjs\hash\build\providers\HashProvider.js
- node_modules\@poppinss\utils\build\src\esmRequire.js
- node_modules\@poppinss\utils\build\index.js
- node_modules\@adonisjs\core\build\src\Ignitor\Bootstrapper\index.js
- node_modules\@adonisjs\core\build\src\Ignitor\Ace\AppCommands.js
- node_modules\@adonisjs\core\build\src\Ignitor\Ace\index.js
- node_modules\@adonisjs\core\build\src\Ignitor\index.js
- ace

```

Any ideas?

**Comment by RomainLanz** [comment, marked as answer]
Hey @WeyIin! 👋 

Few months ago we added a breaking change in the hashing module.
Currently, you should have `@phc/bcrypt` (or something similar) installed, you must remove it and instead use `phc-bcrypt`.

Also, some configurations have change, please, refer to their template in their respective repository.

**Reply by WeyIin** [reply]
Thank you @RomainLanz !

Details of how I solved the issue in case anyone else is having the same problem: 

`npm install --save phc-bcrypt`

Replaced the content of Auth.ts with:
https://github.com/adonisjs/auth/blob/develop/templates/middleware/Auth.txt

Replaced the content of Shield.ts with:
https://github.com/adonisjs/shield/blob/develop/templates/shield.txt
